src/controllers/student.controls.js

```diff
@@ -33,23 +34,43 @@ module.exports = {
   },
   getStudents: async (req, res, next)=>{
     try {
-      const students = await Model.find();
+      const students = await studentModel.find();
       res.status(200).json(students);
     } catch (error) {
       res.status(400).json(error);
     }
   },
   getStudentById: async (req, res, next)=>{
     try {
-      const student = await Model.findOne({studentId: req.params.id});
-      res.status(200).json(student);
+      // const student = await Model.findOne({studentId: req.params.id});
+      studentModel.aggregate([
+        {$lookup: {
+          from: 'courses',
+          localField: 'courses',
+          foreignField: '_id',
+          as: 'courses_data',
+        },
+        },
+        {$match: {
+          studentId: req.params.id,
+        },
```

Comment:
it's usually best to have the `$match` step be the first step in the pipeline. So that you are filtering the returned documents first, before performing more actions on them. Won't make a big difference with your small amount of data, but with 1000s of students, it would.

In this case, you only need to `$lookup` the courses for one student. But in this order, the aggregation is fetching the courses for every student before filtering it down to one

Reply:
so should I remove the $match

Reply:
No, I'm suggesting you put the `$match` step before the `$lookup` step

Reply:
Okay